### PR TITLE
[LoongArch64] Synchronize with PR#103188 to fix the build error.

### PR DESCRIPTION
**Pull request by LuckyXu-HF**
Synchronize with PR https://github.com/dotnet/runtime/pull/103188 to fix the build error on LA64.

### CODE REVIEW

**Comment by LuckyXu-HF** [comment]
@shushanhf @kunalspathak  @jakobbotsch could you please review this PR? Thanks.

File: src/coreclr/jit/targetloongarch64.h [hunk 1]
```diff
@@ -110,7 +110,6 @@
   // Temporary registers used for the GS cookie check.
   #define REG_GSCOOKIE_TMP_0       REG_T0
   #define REG_GSCOOKIE_TMP_1       REG_T1
-  #define RBM_GSCOOKIE_TMP         (RBM_T0|RBM_T1)
```
Comment (kunalspathak) [inline src/coreclr/jit/targetloongarch64.h:113]: why is this deleted?

Comment (LuckyXu-HF) [reply src/coreclr/jit/targetloongarch64.h:113]: Because this is only used in the above change in `lsraloongarch64.cpp`, now we use `~(genSingleTypeRegMask(REG_GSCOOKIE_TMP_0) | genSingleTypeRegMask(REG_GSCOOKIE_TMP_1))` instead, so we can delete this now.

**Review by shushanhf — APPROVED** [review]
LGTM
Thanks!

Hi, @kunalspathak 
Could you please help to review this PR to fix the building errors for LA64?

**Review by kunalspathak — APPROVED** [review]
LGTM
